tests/distgit/mocks.py

```diff
@@ -0,0 +1,62 @@
+from model import Model
+
+class MockDistgit(object):
+    def __init__(self):
+        self.branch = None
+
+
+class MockContent(object):
+    def __init__(self):
+        self.branch = None
+
+
+class MockConfig(object):
+
+    def __init__(self):
+        self.distgit = MockDistgit()
+        self.content = Model()
+        self.content.source = Model()
+        self.content.source.specfile = "test-dummy.spec"
+
+
+class SimpleMockLock(object):
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+class MockRuntime(object):
+
+    def __init__(self, logger):
+        self.branch = None
+        self.distgits_dir = "distgits_dir"
+        self.logger = logger
+        self.mutex = SimpleMockLock()
+        self.missing_pkgs = set()
+
+    def detect_remote_source_branch(self, _):
+        pass
+
+class MockMetadata(object):
+
+    def __init__(self, runtime):
+        self.config = MockConfig()
+        self.runtime = runtime
+        self.logger = runtime.logger
+        self.name = "test"
```

Comment:
I realize that this is just copied over from what we had, and I don't expect you to fix everything in the first pass. But just to set the expectation for how test infra should develop: if at all possible, no one should ever have to look at a mock definition to figure out what they're going to get back from a test. The mock should not fill in values like a fixture (or, if required for functioning at all, these should be explicitly dummy values), and the tests should set any values that they expect to see returned. That way you only have to look at the test method and the code it's testing to understand it.

That's how I summarize the advice in https://testing.googleblog.com/2018/02/testing-on-toilet-cleanly-create-test.html (that's about helper methods but these are essentially helper classes...)

```suggestion
        self.name = None
```

Reply:
:heavy_check_mark: Yes, I'm totally aligned with that idea! No hidden stuff, explicitly setting whatever is needed directly on the test function body is much cleaner.